deployments/server.yaml

```diff
@@ -20,6 +20,10 @@ metadata:
   name: network-resources-injector
   namespace: kube-system
 spec:
+  securityContext:
+    runAsUser: 10000
+    runAsGroup: 10000
+    runAsNonRoot: true
```

Comment:
Does runAsNonRoot require setting nonRoot user in Dockerfile?

Reply:
@zshi-redhat Unfortunately, it seems that it does for NRI. I tried to remove both of the runAsUser and runAsGroup and I got the following error:
```
Error: container has runAsNonRoot and image will run as root
```

Reply:
@martinkennelly IIUC, we need to update Dockerfile to use non-root user, right?
If yes, would you mind updating the Dockerfile in this PR?

Reply:
@zshi-redhat Good idea. I tried it but k8 API doesn't even allow it pass through for scheduling. See the runAsNonRoot validate here: https://github.com/kubernetes/kubernetes/blob/5648200571889140ad246feb82c8f80a5946f167/pkg/kubelet/kuberuntime/security_context.go#L91